near-sdk/src/private/abi.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::HashMap;
+
+use schemars::schema::{RootSchema, Schema};
+use serde::{Deserialize, Serialize};
+
+/// Current version of the ABI schema format.
+const ABI_SCHEMA_SEMVER: &str = "0.1.0";
+
+/// NEAR ABI does not use Rust types, instead it uses abstract type identifiers (represented by
+/// `TypeId`) that are then mapped to the corresponding JSON subschema.
+pub type TypeId = u32;
+
+/// Contract ABI.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct AbiRoot {
+    /// Semver of the ABI schema format.
+    pub abi_schema_version: String,
+    /// Meta information about the contract.
+    pub metainfo: AbiMetainfo,
```

Comment:
do we wanna keep `metainfo` instead of `metadata`?

Reply:
Do you think `metadata` is a better name here? I left `metainfo` because the previous iteration of ABI was called metadata, so no strong opinions here

Reply:
I feel like `metadata` would be more intuitive due to it being used more frequently to denote this kind of info, but if there's metadata being used somewhere else in SDK, then that might be of conflict. But if it's not being used anymore, then it's probably best to just take it

Reply:
Renamed